src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public partial class JsonContent : HttpContent
+    {
+        internal const string JsonMediaType = "application/json";
+        internal const string JsonType = "application";
+        internal const string JsonSubtype = "json";
+        private static MediaTypeHeaderValue s_defaultMediaType
+            => MediaTypeHeaderValue.Parse(string.Format("{0}; charset={1}", JsonMediaType, Encoding.UTF8.WebName));
+
+        internal static JsonSerializerOptions s_defaultSerializerOptions
+            => new JsonSerializerOptions { PropertyNameCaseInsensitive = true, PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+
+        private readonly JsonSerializerOptions? _jsonSerializerOptions;
+        public Type ObjectType { get; }
+        public object? Value { get; }
+
+        private JsonContent(object? inputValue, Type inputType, MediaTypeHeaderValue? mediaType, JsonSerializerOptions? options)
+        {
+            if (inputType == null)
+            {
+                throw new ArgumentNullException(nameof(inputType));
+            }
+
+            if (inputValue != null && !inputType.IsAssignableFrom(inputValue.GetType()))
+            {
+                throw new ArgumentException(SR.Format(SR.SerializeWrongType, inputType, inputValue.GetType()));
+            }
+
+            Value = inputValue;
+            ObjectType = inputType;
+            Headers.ContentType = mediaType ?? s_defaultMediaType;
```

Comment:
This is exposing the s_defaultMediaType singleton as part of the JsonContent headers, and MediaTypeHeaderValue is mutable.  That means if someone constructs a JsonContent and we end up defaulting to this instance, the consumer could then mutate that MediaTypeHeaderValue, and whether they know it or not, they would then be mutating that shared instance for all future uses of that default in the process.

Reply:
It is not a singleton, it is using the expression body syntax, so it should be creating a new instance every time is called.

I am specifically testing for that on https://github.com/dotnet/runtime/pull/33459/files#diff-6029b7b4498ea082a9153f1b1611098fR82-R91

Reply:
Ok, then it shouldn't be named s_defaultMediaType, which conveys that it's a static field. It should instead be DefaultMediaType, or CreateDefaultMediaType(), or something like that.